replica/table.cc

```diff
@@ -2878,16 +2878,17 @@ table::query(schema_ptr s,
     query_state qs(s, cmd, opts, partition_ranges, std::move(accounter));
 
     std::optional<query::querier> querier_opt;
-    if (saved_querier) {
-        querier_opt = std::move(*saved_querier);
+    if (saved_querier && *saved_querier) {
+        querier_opt.emplace(std::move(saved_querier->value()));
+        saved_querier->reset();
```

Comment:
This also doesn't seem to optimize anything. `saved_querier` is always non-null in non-test code.